app/src/main/java/com/github/fribourgsdp/radio/UserProfileActivity.kt

```diff
@@ -76,29 +76,22 @@ class UserProfileActivity : AppCompatActivity() {
 
 
         logoutButton.setOnClickListener{
-            firebaseAuth.signOut()
             startActivity(Intent(this, MainActivity::class.java))
             finish()
         }
 
         googleSignInButton.setOnClickListener{
-            startActivity(Intent(this,GoogleSignInActivity::class.java))
-        }
-
-        userIcon.setColorFilter(PorterDuffColorFilter(user.color, PorterDuff.Mode.ADD))
-
-        findViewById<ImageView>(R.id.userIcon).colorFilter =
-            PorterDuffColorFilter(user.color, PorterDuff.Mode.ADD)
+            if(signedIn){
+                firebaseAuth.signOut()
+                googleSignInButton.setText("Sign in")
```

Comment:
Here, it would be better to use the resources instead of the hardcoded strings.

Reply:
Done